Fix id attribute usage in FormHelper

markstory opened the pull request:
When #16475 was merged a regression was introduced that caused the id values generated for radio options to be wrong when an `id` attribute was set.

Fixes #16835

markstory committed:
Fix id attribute usage in FormHelper

When #16475 was merged a regression was introduced that caused the id
values generated for radio options to be wrong when an `id` attribute
was set.

Fixes #16835

ndm2 commented:
Does the same problem maybe apply to `multiCheckbox()` too, given that it is doing the same unsetting?

https://github.com/cakephp/cakephp/blob/3efb9647e0325260e8f1018dae4d00bcf056366a/src/View/Helper/FormHelper.php#L2143-L2145

markstory commented:
> Does the same problem maybe apply to multiCheckbox() too, given that it is doing the same unsetting?

It could, I'll add more tests to see if the same problem applies there.

markstory committed:
Fix custom id problem in multiCheckbox()

Much like the problem in radio() this method is also impacted. Shout out
to @ndm2 who identified this issue during review.

markstory committed:
Fix phpcs
